File: java/program/BarangElektronik.java

import java.util.Map;
import java.text.DecimalFormat;
// Class Level 1: BarangElektronik
class BarangElektronik {
    protected int id;
    protected String nama;
    protected String merek;
    protected double harga;

    public BarangElektronik(int id, String nama, String merek, double harga) {
        this.id = id;
        this.nama = nama;
        this.merek = merek;
        this.harga = harga;
    }

    // Getters
    public int getId() { return id; }
    public String getNama() { return nama; }
    public String getMerek() { return merek; }
    public double getHarga() { return harga; }

    // Setters
    public void setNama(String nama) { this.nama = nama; }
    public void setMerek(String merek) { this.merek = merek; }
    public void setHarga(double harga) { this.harga = harga; }

    public void display(Map<String, Integer> colWidths) {
        DecimalFormat df = new DecimalFormat("#.00");
        String formattedHarga = df.format(getHarga());
        System.out.printf("| %-" + colWidths.get("ID") + "s | %-" + colWidths.get("Nama") + "s | %-" + colWidths.get("Merek") + "s | %-" + colWidths.get("Harga") + "s |",  getId(), getNama(), getMerek(), formattedHarga);
    }
}

File: python/program/BarangElektronik.py
# Class Level 1: BarangElektronik

class BarangElektronik:
    def __init__(self, id, nama, merek, harga):
        self.id = id
        self.nama = nama
        self.merek = merek
        self.harga = harga

    def get_id(self):
        return self.id
    def get_nama(self):
        return self.nama
    def get_merek(self):
        return self.merek
    def get_harga(self):
        return self.harga
    
    def set_nama(self, nama):
        self.nama = nama
    def set_merek(self, merek):
        self.merek = merek
    def set_harga(self, harga):
        self.harga = harga

    def display(self, col_widths):
        return f"| {self.get_id():<{col_widths['ID']}} | {self.get_nama():<{col_widths['Nama']}} | {self.get_merek():<{col_widths['Merek']}} | {self.get_harga():<{col_widths['Harga']}.2f} |"